### PR TITLE
added CompositeField and all related goodies

### PR DESCRIPTION
**Pull request by miki725**

### CODE REVIEW

File: django_sorcery/db/mixins.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        Should raise ``ValidationError`` if any errors are found.
+        """
+
+    def clean_fields(self, exclude):
+        """
+        Clean all fields on object
+        """
+        errors = {}
+
+        # import pdb; pdb.set_trace()
```
Comment (shosca) [inline django_sorcery/db/mixins.py]: Oops :trollface:

File: Makefile [hunk 2]
```diff
@@ -41,13 +41,13 @@ coverage: ## check code coverage quickly with the default Python
 		--cov-report term-missing \
 		--cov=django_sorcery tests \
 		--doctest-modules \
-		django_sorcery tests
+		tests django_sorcery
```
Comment (miki725) [inline Makefile]: tests initialize django so it needs to be first. or need to use pytest-django

Comment (shosca) [reply Makefile]: :+1:

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -15,6 +15,12 @@
   - id: flake8
   - id: trailing-whitespace
 
+- repo: https://github.com/miki725/importanize
```
Comment (shosca) [inline .pre-commit-config.yaml]: Oh you..

Comment (miki725) [reply .pre-commit-config.yaml]: yup

**Pull request merged**